src/components/HeaderSearch/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect, createRef } from 'react'
+import PropTypes from 'prop-types'
+import { Input, Icon, AutoComplete } from 'antd'
+import classNames from 'classnames'
+import Debounce from 'lodash-decorators/debounce'
+import Bind from 'lodash-decorators/bind'
+
+const styles = require('./index.less')
+
+// TODO: define on next task
+// static propTypes = {
+//   className: PropTypes.string,
+//   placeholder: PropTypes.string,
+//   onSearch: PropTypes.func,
+//   onPressEnter: PropTypes.func,
+//   defaultActiveFirstOption: PropTypes.bool,
+//   dataSource: PropTypes.array,
+//   defaultOpen: PropTypes.bool,
+//   onVisibleChange: PropTypes.func
+// }
+
+// static defaultProps = {
+//   defaultActiveFirstOption: false,
+//   onPressEnter: () => {},
+//   onSearch: () => {},
+//   className: '',
+//   placeholder: '',
+//   dataSource: [],
+//   defaultOpen: false,
+//   onVisibleChange: () => {}
+// }
```

Comment:
I  think you don't need `PropTypes` here. I would just use TypeScript types.

Reply:
👍